resources/user.go

```diff
@@ -0,0 +1,246 @@
+package resources
+
+import (
+	"context"
+	"errors"
+	models "github.com/zededa/terraform-provider/models"
+	zschema "github.com/zededa/terraform-provider/schemas"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	api_client "github.com/zededa/terraform-provider/client"
+	identity_access_management "github.com/zededa/terraform-provider/client/identity_access_management"
+)
+
+/*
+User management API
+*/
+
+func UserResource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: IdentityAccessManagement_CreateUser,
+		DeleteContext: IdentityAccessManagement_DeleteUser,
+		ReadContext:   IdentityAccessManagement_GetUser,
+		UpdateContext: IdentityAccessManagement_UpdateUser2,
```

Comment:
any reason to name the function IdentityAccessManagement_UpdateUser2 instead of IdentityAccessManagement_UpdateUser ?

Reply:
Code generator created two versions: IdentityAccessManagement_UpdateUser and IdentityAccessManagement_UpdateUser2. IdentityAccessManagement_UpdateUser2 is enhanced version so i used this and removed not used IdentityAccessManagement_UpdateUser. Maybe in the future we would need the version with name IdentityAccessManagement_UpdateUser but not now.